packages/analytics/client/src/analytics_client/analytics_client.ts

```diff
@@ -360,3 +360,39 @@ export class AnalyticsClient implements IAnalyticsClient {
       });
   }
 }
+
+/** security paths that contain user data */
+const SECURITY_PATHS = ['hosts/name', 'users/name', 'network/ip'];
+// this indicates a user query
+const queryMarker = `?_g=`;
+const fixUrl = (url: string) => {
+  if (url.includes(queryMarker)) {
+    url = url.substring(0, url.indexOf(queryMarker) + queryMarker.length);
+  }
+  const matchedKnownPiiData = SECURITY_PATHS.find((path) => url.includes(path));
+
+  if (matchedKnownPiiData) {
+    const res = url.split(matchedKnownPiiData);
+    return res[0] + matchedKnownPiiData + res[1].replace(/^\/[^\/\r\n]+(?=\/)/, '/MASKED');
+  }
+
+  return url;
+};
+
+export function maskSecurityUrls(properties: Partial<EventContext>): Record<string, unknown> {
+  const maskedProperties: Partial<EventContext> = {};
+  if (properties.page_url) {
+    maskedProperties.page_url = fixUrl(properties.page_url as string);
```

Comment:
`page` and `pageName` are populated here: https://github.com/elastic/kibana/blob/4c626f16c0bfc333bf0591f586b082f6c09dd3b1/packages/core/execution-context/core-execution-context-browser-internal/src/execution_context_service.ts#L124-L129

TBH, bearing in mind the source of that information (execution context), it shouldn't expose any PII. `page` is provided by the app/plugin. After double-checking our telemetry data, we can confirm these usages for the field:
* Some plugins provide human-readable indicators of subsections of the app ("app", "listing", "editor", "console")
	* Special mention to `management` that shows the name of each section in the Stack Management view "ingestPipelines", "watcher", "indexManagement", etc. 
	* `visualize` also shows nested values to the editor value to indicate the type of viz that's being used ("editor:data_table", "editor:tsvb", "editor:vertical_bar", ...)
* Others like Security Solution, show the URL path. I can see versions prior to 8.8.0 show generic URLs like `/rules/id/:detailName/:tabName(rule_exceptions)`. After 8.8.0, they show the actual values. I wonder if this PR introduced the regression: https://github.com/elastic/kibana/pull/152816

FWIW, the execution context [already carries the pathname](https://github.com/elastic/kibana/blob/4c626f16c0bfc333bf0591f586b082f6c09dd3b1/packages/core/execution-context/core-execution-context-browser-internal/src/execution_context_service.ts#L103) (intentionally not used in telemetry, only for logging purposes to make troubleshooting easier). So I wonder if we should fix the `page` reported by Security Solutions instead of "polluting" the EBT client with this filter.

To be clear: I'm not opposed to running PII filters in the EBT client. FWIW, I think we should traverse the entire object to ensure we don't leak any PII anywhere (things like the one we are looking at may occur) https://github.com/elastic/kibana/issues/132259. However, for some reason, patching these fields here instead of fixing the source of the leak seems wrong. I think it's mostly because we are implementing the filter in a reactive way: if a new URL is added `nodes/name/:piiSensitiveName`, we'll have to remember to add it to the list, or we'll leak those values in the versions released until we fix them.

At the same time, the Referrer header (refer to [my previous comment](https://github.com/elastic/kibana/pull/164378#pullrequestreview-1588830835)) keeps coming to mind as a reminder of "is it really worth the effort of hiding those here?".

Fixing the source of `page` (and hence `pageName`) would improve analytics, as we'll be able to see all events in page `/rules/id/:detailName/:tabName(alerts)`, rather than `/rules/id/387817b0-2fb3-11ee-9a91-19aeacd9e5a8/alerts`. We could refer to `page_url` to identify if they spend most of the time looking at the same rule/host or if the cardinality is high.

WDYT?